src/App.jsx

```diff
@@ -14,45 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortFiled, setSortFiled] = useState('');
+  const [reverse, setReverse] = useState('');
+
+  function sortAlphabetically() {
+    if (reverse === 'R') {
+      setVisibleGoods([...visibleGoods].sort().reverse());
+    } else {
+      setVisibleGoods([...visibleGoods].sort());
+    }
+
+    setSortFiled('A');
+  }
+
+  function sortByLength() {
+    if (reverse === 'R') {
+      setVisibleGoods([...visibleGoods].sort(
+        (good1, good2) => good2.length - good1.length,
+      ));
+    } else {
+      setVisibleGoods([...visibleGoods].sort(
+        (good1, good2) => good1.length - good2.length,
+      ));
+    }
+
+    setSortFiled('L');
+  }
+
+  function reverseSort() {
+    setVisibleGoods([...visibleGoods].reverse());
+    if (reverse === '') {
+      setReverse('R');
+    } else {
+      setReverse('');
+    }
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function reset() {
+    setVisibleGoods(goodsFromServer);
+    setSortFiled('');
+    setReverse('');
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={sortAlphabetically}
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortFiled !== 'A' },
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={sortByLength}
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortFiled !== 'L' },
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={reverseSort}
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': reverse !== 'R' },
+          )}
+        >
+          Reverse
+        </button>
+
+        {JSON.stringify(goodsFromServer) !== JSON.stringify(visibleGoods)
```

Comment:
```suggestion

```
You can check the state of the component here

Reply:
Fix it, you can just check if your sortFiled in the state is not empty so that`s means you have some changes

Reply:
then if i press reverse, clear button doesn't appear